source/blog/2018-03-06-release-notes.md

```diff
@@ -0,0 +1,190 @@
+---
+layout: post
+title: 🌷 March 2018 Release Notes - Version 18.3.0
+published: true
+post_author:
+  display_name: Stefan Lance
+  twitter: stefan_lance
+tags: Releases, Finagle, Finatra, Util, Scrooge, TwitterServer
+---
+
+Our March releases have arrived!
+
+[Finagle 18.3.0][finagle], [Finatra 18.3.0][finatra], [Scrooge 18.3.0][scrooge], [TwitterServer 18.3.0][twitterserver], and [Util 18.3.0][util].
+
+### [Finagle](https://github.com/twitter/finagle/) ###
+
+New Features
+
+  * finagle-core: `c.t.f.client.BackupRequestFilter.filterService` for wrapping raw services in a
+    `c.t.f.client.BackupRequestFilter` is now public. [b227d988](https://github.com/twitter/finagle/commit/b227d98821ae5aa7bfa39f5c46b354e3407204d1)
+
+  * finagle-core: Introduce `c.t.f.Stacks.EMPTY_PARAMS` for getting an empty Param map from
+    Java, and `c.t.f.Stack.Params.plus` for easily adding Params to a Param map from Java.
+    [43e0f007](https://github.com/twitter/finagle/commit/43e0f007a18f923c4280b7e2746952f2986eac28)
+
+Bug Fixes
+
+  * finagle-core: `c.t.f.liveness.FailureAccrualFactory` takes no action on `c.t.f.Failure.Ignorable`
+    responses. [512894cf](https://github.com/twitter/finagle/commit/512894cf5b98b7d7b5e7ec7261a7db447a8e7f06)
+
+  * finagle-core: `c.t.f.pool.WatermarkPool` is resilient to multiple closes on a service instance.
+    [37c29e9e](https://github.com/twitter/finagle/commit/37c29e9e06dbecac64ffc97385e6f1bde4b5ff7e)
+
+  * finagle-core: `c.t.f.pool.CachingPool` service wrapper instances are resilient to multiple closes.
+    [ce779ec7](https://github.com/twitter/finagle/commit/ce779ec7f8ab0fcf39a8c0e9797af2b074214777)
+
+  * finagle-core: Requeue module now closes sessions it prevented from propagating up the stack.
+    [f5cdda14](https://github.com/twitter/finagle/commit/f5cdda14cf5675eb4f0fea230b8efb1ab7bbff95)
+
+  * finagle-base-http: `c.t.f.http.Netty4CookieCodec.encode` now wraps Cookie values that would
+    be wrapped in `c.t.f.http.Netty3CookieCodec.encode`. [78fdc9aa](https://github.com/twitter/finagle/commit/78fdc9aa62c55cc73bf7fe78a0d7c0b7376f5567)
+
+  * finagle-base-http: `c.t.f.http.Cookie.maxAge` returns `c.t.f.http.Cookie.DefaultMaxAge`
+    (instead of null) if maxAge has been set to null or None in the copy constructor
+    [17a32d44](https://github.com/twitter/finagle/commit/17a32d4461929a20e543117c9c0c82c2cb8159a1).
+
+  * finagle-http: The HTTP client will not attempt to retry nacked requests with streaming
+    bodies since it is likely that at least part of the body was already consumed and therefore
+    it isn't safe to retry. [a787955b](https://github.com/twitter/finagle/commit/a787955bc277a9f6928c29a05bb5ee7ddc3185df)
+
+Breaking API Changes
+
+  * finagle-base-http: Removed `c.t.f.http.Cookie.comment_`, `c.t.f.http.Cookie.comment_=`,
+    `c.t.f.http.Cookie.commentUrl_`, and `c.t.f.http.Cookie.commentUrl_=`. `comment` and `commentUrl`
+    per RFC-6265. NOT FOUND: D137538
```

Comment:
whoops, i think this is https://github.com/twitter/finagle/commit/13a6bd2c75339397eb8dbada68dcf0a85d6529aa.

Reply:
Nice catch! Maybe I should update the script to crash if it can't find one!

Reply:
I think that would be great!

Reply:
Agreed. Great eyes, @yufangong 😃